packages/patterns/src/components/pattern-overrides-controls.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useId } from '@wordpress/element';
+import { InspectorControls } from '@wordpress/block-editor';
+import { ToggleControl, BaseControl, Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PARTIAL_SYNCING_SUPPORTED_BLOCKS,
+	PATTERN_OVERRIDES_BINDING_SOURCE,
+} from '../constants';
+import AllowOverridesModal from './allow-overrides-modal';
+
+function removeBindings( bindings, syncedAttributes ) {
+	let updatedBindings = {};
+	for ( const attributeName of syncedAttributes ) {
+		// Omit any bindings that's not the same source from the `updatedBindings` object.
+		if (
+			bindings?.[ attributeName ]?.source !==
+				PATTERN_OVERRIDES_BINDING_SOURCE &&
+			bindings?.[ attributeName ]?.source !== undefined
+		) {
+			updatedBindings[ attributeName ] = bindings[ attributeName ];
+		}
+	}
+	if ( ! Object.keys( updatedBindings ).length ) {
+		updatedBindings = undefined;
+	}
+	return updatedBindings;
+}
+
+function addBindings( bindings, syncedAttributes ) {
+	const updatedBindings = { ...bindings };
+	for ( const attributeName of syncedAttributes ) {
+		if ( ! bindings?.[ attributeName ] ) {
+			updatedBindings[ attributeName ] = {
+				source: PATTERN_OVERRIDES_BINDING_SOURCE,
+			};
+		}
+	}
+	return updatedBindings;
+}
+
+function PatternOverridesControls( { attributes, name, setAttributes } ) {
+	const controlId = useId();
+	const [ showAllowOverridesModal, setShowAllowOverridesModal ] =
+		useState( false );
+
+	const syncedAttributes = PARTIAL_SYNCING_SUPPORTED_BLOCKS[ name ];
+	const attributeSources = syncedAttributes.map(
+		( attributeName ) =>
+			attributes.metadata?.bindings?.[ attributeName ]?.source
+	);
+	const isConnectedToOtherSources = attributeSources.every(
+		( source ) => source && source !== 'core/pattern-overrides'
+	);
+
+	function updateBindings( isChecked, customName ) {
+		if ( isChecked && ! attributes.metadata?.name && ! customName ) {
+			setShowAllowOverridesModal( true );
+			return;
+		}
```

Comment:
To me it feels like this isn't needed. Is there ever a case where you toggle the `ToggleControl` and have the modal shown?

As noted in the other comment, perhaps the button could call `setShowAllowOverridesModal( true )` directly.

Reply:
Oops, yeah. I think it's from an older exploration.